docker/Dockerfile.builder

```diff
@@ -1,27 +1,39 @@
+# syntax=docker/dockerfile:1.7-labs
+
 # Target with dependencies to build all flow tools from their sources.
 # i.e., "./build_openroad.sh --local" from inside a docker container
 # NOTE: don't use this file directly unless you know what you are doing,
 # instead use etc/DockerHelper.sh
+
 ARG fromImage=openroad/flow-ubuntu22.04-dev:latest
 
-FROM $fromImage AS openroad-builder-base
+FROM $fromImage AS orfs-base
 
-ARG numThreads=$(nproc)
-
-COPY . /OpenROAD-flow-scripts
 WORKDIR /OpenROAD-flow-scripts
+COPY --link dev_env.sh dev_env.sh
+COPY --link build_openroad.sh build_openroad.sh
 
-RUN ./build_openroad.sh --no_init --local --threads ${numThreads}
-
-FROM $fromImage AS openroad-flow-scripts-base
+FROM orfs-base AS orfs-builder-base
 
-COPY . /OpenROAD-flow-scripts
-
-RUN rm -rf /OpenROAD-flow-scripts/tools /OpenROAD-flow-scripts/.git
-
-COPY --from=openroad-builder-base /OpenROAD-flow-scripts/tools/install /OpenROAD-flow-scripts/tools/install
-
-FROM $fromImage
+COPY --link tools tools
+ARG numThreads=$(nproc)
 
-COPY --from=openroad-flow-scripts-base /OpenROAD-flow-scripts /OpenROAD-flow-scripts
-WORKDIR /OpenROAD-flow-scripts
+RUN echo "" > tools/yosys/abc/.gitcommit && \
+  ./build_openroad.sh --no_init --local --threads ${numThreads}
+
+FROM orfs-base
+
+COPY --link flow/designs flow/designs
+COPY --link flow/platforms flow/platforms
```

Comment:
sort these by descending update size/frequency in git history. Especially platforms is important to have at the top, it is big and rarely changes.